lisa/sut_orchestrator/azure/common.py

```diff
@@ -1904,3 +1916,272 @@ def get_size(disk_type: schema.DiskType, data_disk_iops: int = 1) -> int:
             return iops_dict[min_iops]
         else:
             raise LisaException(f"Data disk type {disk_type} is unsupported.")
+
+
+def add_system_assign_identity(
+    platform: "AzurePlatform",
+    resource_group_name: str,
+    vm_name: str,
+    location: str,
+    log: Logger,
+) -> Any:
+    compute_client = ComputeManagementClient(
+        platform.credential, platform.subscription_id
+    )
+    params_identity = {"type": "SystemAssigned"}
+    params_create = {"location": location, "identity": params_identity}
+
+    try:
+        vm_poller = compute_client.virtual_machines.begin_create_or_update(
+            resource_group_name,
+            vm_name,
+            params_create,
+        )
+        vm_result = vm_poller.result()
+        object_id_vm = vm_result.identity.principal_id
+        log.info(object_id_vm)
+
+        if not object_id_vm:
+            raise ValueError("object_id_vm is not set.")
+
+        return object_id_vm
+
+    except Exception as e:
+        log.error(f"Error creating VM: {e}")
+        raise LisaException(str(e))
+
+
+def get_key_vault_management_client(
+    platform: "AzurePlatform",
+) -> KeyVaultManagementClient:
+    return KeyVaultManagementClient(platform.credential, platform.subscription_id)
+
+
+def create_keyvault(
+    platform: "AzurePlatform",
+    resource_group_name: str,
+    tenant_id: str,
+    object_id: str,
+    object_id_vm: str,
+    location: str,
+    vault_name: str,
+) -> Any:
+    keyvault_client = get_key_vault_management_client(platform)
+
+    vault_properties = VaultProperties(
+        tenant_id=tenant_id,
+        sku=KeyVaultSku(name="standard"),
+        access_policies=[
+            AccessPolicyEntry(
+                tenant_id=tenant_id,
+                object_id=object_id,
+                permissions=Permissions(
+                    keys=["all"], secrets=["all"], certificates=["all"]
+                ),
+            ),
+            AccessPolicyEntry(
```

Comment:
The access policy of the machine should be in a separated method. In this method, it creates the keyvault, so it shouldn't do extra work, like assign policy to a vm. So this method should only assign policy to the current account for convenient.